lib/Mail/Milter/Authentication/Metric.pm

```diff
@@ -125,10 +125,21 @@ sub prom {
     if ( ! -e $metric_tempfile ) {
         $prom = undef;
     }
+    if ( ! -d $metric_tempfile ) {
+        # If metric_tempfile is a regular file then we need to re-init with a directory
+        # this is likely a restart after upgrade.
+        $prom = undef;
+    }
 
     if ( ! $prom ) {
+        if ( -f $metric_tempfile ) {
+            unlink $metric_tempfile;
+        }
+        if ( ! -d $metric_tempfile ) {
```

Comment:
You actually don't need this. `Hash::SharedMem` will create it. In fact it's probably better that you don't, in order to preserve the abstraction.

(but you will maybe tell me "we have to because X", and I'm interested in that!)

Reply:
explained, we create the metrics in a subdir with a known name for extra safety. We don't want to accidentally remove a directory on init if the user misconfigures.